### PR TITLE
clean up parsing full system

### LINKED ISSUE #833: Clean Up Argument Parsing In full_system
### Description of the task
Currently `full_system_main.cpp` is a  bit messy, and in particular argument parsing could probably be made a little cleaner. We should clean it up!

### Acceptance criteria

- [ ] `full_system_main.cpp` is cleaned up!

### PR DESCRIPTION
**Pull request by LiCody**
### Description
Cleaned up parsing by:
- deleting global variable "headless"
- making parseCommandLineArgs return a struct "args" of command line inputs
- handling parts of said struct in main  

### Testing Done
ran file with bazel with combinations of arguments, --help, --backend grsim, --backend grsim --headless, --help --backend grsim.

### Resolved Issues
resolves #833 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/full_system_main.cpp [hunk 1]
```diff
@@ -56,53 +62,50 @@ void setBackendFromString(std::string backend_name)
...
 
-    return true;
+    return args;
 }
 
 /**
  * Connects all the observers together
  */
-void connectObservers()
+void connectObservers(bool headless)
```
Comment (matthewberends) [inline src/software/full_system_main.cpp:111]: Perhaps we could add to the doc comment to give a quick description of what the headless param is for.

File: src/software/full_system_main.cpp [hunk 2]
```diff
@@ -56,53 +62,50 @@ void setBackendFromString(std::string backend_name)
  *
  * @return True if the program should continue, false otherwise
```
Comment (matthewberends) [inline src/software/full_system_main.cpp]: Update this comment now that the return value is different

**Review by matthewberends — CHANGES_REQUESTED** [review]
Looking good to me. Just a couple of small documentation comments

**Review by garethellis0 — COMMENTED** [review]
Nothing to add on top of Matthews comments

File: src/software/full_system_main.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/full_system_main.cpp:111]: Headless param is for whether or not we run the Visualizer. if `headless=true` the Visualizer is not run. If `headless=false`, the Visualizer is run.

**Review by matthewberends — DISMISSED** [review]
LGTM!

**Comment by garethellis0** [comment]
looks like you need to fix formatting. you can do so by running <pre>./clang_format/fix_formatting.sh -b master</pre> from the `Software` folder

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**